DigitalLearningSolutions.Web/Controllers/TrackingSystem/CentreConfiguration/RegistrationPromptsController.cs

```diff
@@ -71,32 +90,83 @@ public IActionResult EditRegistrationPrompt(EditRegistrationPromptViewModel mode
 
         [HttpGet]
-        [Route("Add/SelectPrompt")]
-        public IActionResult AddRegistrationPromptSelectPrompt()
+        [Route("{promptNumber:int}/Edit/Bulk")]
+        [ServiceFilter(typeof(RedirectEmptySessionData<EditRegistrationPromptData>))]
+        public IActionResult EditRegistrationPromptBulk(int promptNumber)
         {
-            var addRegistrationPromptData = TempData.Peek<AddRegistrationPromptData>();
+            var data = TempData.Peek<EditRegistrationPromptData>()!;
 
-            if (addRegistrationPromptData == null || !Request.Cookies.ContainsKey(CookieName))
+            var model = new BulkRegistrationPromptAnswersViewModel(
+                data.EditModel.OptionsString,
+                false,
+                promptNumber
+            );
+
+            return View("BulkRegistrationPromptAnswers", model);
+        }
+
+        [HttpPost]
+        [Route("Bulk")]
+        [ServiceFilter(typeof(RedirectEmptySessionDataXor<AddRegistrationPromptData, EditRegistrationPromptData>))]
+        public IActionResult RegistrationPromptBulkPost(BulkRegistrationPromptAnswersViewModel model)
+        {
+            ValidateBulkOptionsString(model.OptionsString);
+            if (!ModelState.IsValid)
             {
-                addRegistrationPromptData = new AddRegistrationPromptData();
-                var id = addRegistrationPromptData.Id;
-
-                Response.Cookies.Append(
-                    CookieName,
-                    id.ToString(),
-                    new CookieOptions
-                    {
-                        Expires = DateTimeOffset.UtcNow.AddDays(1)
-                    }
-                );
-                TempData.Set(addRegistrationPromptData);
+                return View("BulkRegistrationPromptAnswers", model);
+            }
+
+            if (model.IsAddPromptJourney)
+            {
+                var addData = TempData.Peek<AddRegistrationPromptData>()!;
+                addData.ConfigureAnswersViewModel!.OptionsString = model.OptionsString;
+                TempData.Set(addData);
+
+                return RedirectToAction("AddRegistrationPromptConfigureAnswers");
             }
 
+            var editData = TempData.Peek<EditRegistrationPromptData>()!;
+            editData.EditModel!.OptionsString = model.OptionsString;
+            TempData.Set(editData);
+
+            return RedirectToAction("EditRegistrationPrompt", new { promptNumber = model.PromptNumber });
+        }
+
+        [HttpGet]
+        [Route("Add/New")]
+        public IActionResult AddRegistrationPromptNew()
+        {
+            TempData.Clear();
+
+            var addRegistrationPromptData = new AddRegistrationPromptData();
+            var id = addRegistrationPromptData.Id;
+
+            Response.Cookies.Append(
+                CookieName,
+                id.ToString(),
+                new CookieOptions
+                {
+                    Expires = DateTimeOffset.UtcNow.AddDays(1)

```

Comment:
Could we pull the cookie expiry time in days to the top of the file and use it for both add & edit? (then it can be changed easily too)
Also, I now wonder whether we should increase the time a bit as we clear it at the beginning of the journey anyway (I'm thinking of maybe a week)🤔